trailblazer/services/encryption_service/utils.py

```diff
@@ -0,0 +1,35 @@
+import base64
+import os
+from cryptography.hazmat.primitives.ciphers.aead import AESGCM
+
+NONCE_SIZE = 12
+
+
+def encrypt_data(data: str, secret_key: str) -> str:
+    data: bytes = data.encode()
+    cipher_text: bytes = _encrypt_with_aes(data=data, secret_key=secret_key)
+    return _bytes_to_string(cipher_text)
+
+
+def decrypt_data(data: str, secret_key: str) -> str:
+    data: bytes = base64.b64decode(data)
+    plain_text: bytes = _decrypt_with_aes(data=data, secret_key=secret_key)
+    return plain_text.decode()
+
+
+def _encrypt_with_aes(data: bytes, secret_key: str) -> bytes:
+    cipher = AESGCM(base64.b64decode(secret_key))
+    nonce: bytes = os.urandom(NONCE_SIZE)
```

Comment:
A bit uncertain how the os.urandom works, but does it give a random byte string _up to_ 12 in length or exactly 12? If the former I think the decryption will fail since the length can vary

Reply:
Exactly 12! So it won't be an issue (this module is covered by tests as well)